tool_helper_utils/__init__.py

```diff
@@ -2,12 +2,30 @@
 import sys
 
 
+class AbstractBaseFactoryFactory:
+    def build():
+        return None
+
+
+class FactoryFactory(AbstractBaseFactoryFactory):
+    def __init__(self, factory):
+        self.factory = factory
+
+    def build(self):
+        return self.factory.get_bad_idea()
```

Comment:
This seems unnecessarily limiting. The factory could build much more than just bad ideas in the future.

Maybe for now it would be enough to just choose a less restrictive method name, like `get_factory_product` (or rather `maybe_get_factory_product` or `get_factory_product_or_status` if we want to allow for error conditions in the factory).